gem/impero_utils.py

```diff
@@ -41,6 +41,18 @@ def preprocess_gem(expressions):
     return expressions
 
 
+def optimise_gem(expressions, quadrature_indices, argument_indices):
+    """Optimisation pass for gem expressions. Optimisation include:
+    - replace division with multiplication
+    - loop invariant code motion
+    - pre-evaluation
+    """
+    expressions = optimise.replace_division(expressions)
+    expressions = optimise.optimise_expressions(
+        expressions, tuple(quadrature_indices), argument_indices)
+    return expressions
```

Comment:
Sorry, I see now this is no longer changing `preprocess_gem`. Anyway, this function shall not be in `gem.impero_utils`.